### ISSUE TITLE
Changing manager of a department requires more than 2G of memory

### ISSUE THREAD

**Issue by iprok** [osTicket/osTicket#5428]
### Prerequisites

* [ ] Can you reproduce the problem in a fresh installation of the "develop" branch?
Haven't tried.

* [x] Do you have any errors in the PHP error log, or javascript console?

`Mar 23 21:57:37 help ool help: PHP Fatal error:  Allowed memory size of 2147483648 bytes exhausted (tried to allocate 20480 bytes) in /var/www/help/include/class.orm.
php on line 3478`

* [x] Did you check the [osTicket forums](https://forum.osticket.com/)?
Yes

* [x] Did you [perform a cursory search](https://github.com/osTicket/osTicket/issues) to see if your bug or enhancement is already reported?

Yes

### Description

Changing the Manager of "Archive Requests" department takes to much time and memory.

### Steps to Reproduce

1. Open osticket/scp/departments.php?id=15 (id=15 is "Archive Requests" in our system)

2. Change "Manager" to any other. I've tried changing from current value to None.

3. Save changes.

**Expected behavior:** [What you expected to happen]

Changes are applied quickly and correctly.

**Actual behavior:** [What actually happened]

Waiting for a long time (more than 20 seconds) then 500 error with "Allowed memory size exhausted error in logs"

### Versions

osTicker 1.14.1
Debian Stretch
PHP 7.0.33-0+deb9u7

I've tried memory_limit setting from 128M till 2GB without any success. We have approx 7000 tickets in our system.

**Comment by JediKev** [comment]
@iprok 

That shouldn't happen. I cannot replicate the same issues with v1.14.1 and PHP 7.3. Take a look at your error logs (Apache Error Logs, PHP Error Logs, MySQL Error Logs, osTicket System Logs, etc.) to see if you can find any related errors.

Cheers.

**Comment by iprok** [comment]
Hello! I've already mentioned the message from logs that is related to the situation:
`Mar 23 21:57:37 help ool help: PHP Fatal error: Allowed memory size of 2147483648 bytes exhausted (tried to allocate 20480 bytes) in /var/www/help/include/class.orm. php on line 3478`

**Comment by JediKev** [comment]
@iprok 

That's not a full stack trace so it's really not that helpful. All that says is the memory was exhausted whilst running the `_cast()` method in class ORM, but this does not tell me anything more.

Cheers.

**Comment by iprok** [comment]
Here is stack trace (from php slow log):
```
[23-Mar-2020 22:38:07]  [pool help] pid 29389
script_filename = /var/www/help/scp/departments.php
[0x00007f69f1553c50] query() /var/www/help/include/mysqli.php:188
[0x00007f69f1553b40] db_query() /var/www/help/include/class.orm.php:3459
[0x00007f69f1553aa0] execute() /var/www/help/include/class.orm.php:3506
[0x00007f69f1553a10] getArray() /var/www/help/include/class.orm.php:1999
[0x00007f69f1553950] {closure}() /var/www/help/include/class.orm.php:2043
[0x00007f69f15538b0] next() /var/www/help/include/class.orm.php:2022
[0x00007f69f1553840] rewind() /var/www/help/include/class.orm.php:2031
[0x00007f69f15537c0] valid() /var/www/help/include/class.orm.php:1703
[0x00007f69f1553720] fillTo() /var/www/help/include/class.orm.php:1713
[0x00007f69f15536b0] asArray() /var/www/help/include/class.orm.php:1336
[0x00007f69f1553640] all() /var/www/help/include/class.orm.php:1359
[0x00007f69f15535a0] one() /var/www/help/include/class.orm.php:606
[0x00007f69f15534a0] lookup() /var/www/help/include/class.dept.php:574
[0x00007f69f1553420] getParent() /var/www/help/include/class.dept.php:593
[0x00007f69f1553380] getFullPath() /var/www/help/include/class.dept.php:594
[0x00007f69f15532e0] getFullPath() /var/www/help/include/class.dept.php:594
[0x00007f69f1553240] getFullPath() /var/www/help/include/class.dept.php:594
[0x00007f69f15531a0] getFullPath() /var/www/help/include/class.dept.php:594
[0x00007f69f1553100] getFullPath() /var/www/help/include/class.dept.php:594
[0x00007f69f1553060] getFullPath() /var/www/help/include/class.dept.php:594
```

**Comment by iprok** [comment]
I got more details. The issue only happens if Parent of the department is set to itself (look at screenshot) AND changing the Manager. I don't know how the Parent of the department was set to himself but after changing to "Top-level department" issue has gone away. Seems that the only problem is that osticket allows such erroneous setting.

**Comment by JediKev** [comment]
@iprok 

Yes, we should not allow you to set the parent to itself. Thank you for the report, we will look into this.

Cheers.

**Comment by JediKev** [comment]
@iprok 

Here you go:
- https://github.com/osTicket/osTicket/pull/5429

Cheers.
